packages/utils/upgrade/src/modules/upgrader/upgrader.ts

```diff
@@ -236,32 +234,12 @@ export class Upgrader implements UpgraderInterface {
   }
 
   private async runCodemods(range: Version.Range): Promise<void> {
-    const repository = codemodRepositoryFactory(
-      codemodRepositoryConstants.INTERNAL_CODEMODS_DIRECTORY
-    );
-
-    // Make sure we have access to the latest snapshots of codemods on the system
-    repository.refresh();
-
-    const versionedCodemods = repository.findByRange(range);
-
-    const hasCodemodsToRun = versionedCodemods.length > 0;
-    if (!hasCodemodsToRun) {
-      this.logger?.debug(`Found no codemods to run for ${f.version(this.target)}`);
-      return;
+    const codemodRunner = codemodRunnerFactory(this.project, range);
+    codemodRunner.dry(this.isDry);
+    if (this.logger) {
```

Comment:
You don't need the if, `logger` has the same type in both `Upgrader` and `CodemodRunner`. You can chain it to the `dry()` call

Reply:
The logger can be null here